src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEnumerator.Windows.cs

```diff
@@ -239,13 +238,12 @@ private void InternalDispose(bool disposing)
                         _pending = null;
                     }
 
-                    if (_pinnedBuffer.IsAllocated)
-                        _pinnedBuffer.Free();
-
-                    if (_buffer != null)
-                        ArrayPool<byte>.Shared.Return(_buffer);
+                    if (_buffer != default)
+                    {
+                        Marshal.FreeHGlobal(_buffer);
```

Comment:
Is it possible anything above might throw and cause us to not get here?

Is it possible a race / misuse of the public Dispose could cause this pointer to be double-freed?

Reply:
CloseHandle won't throw and we lock here so I don't think we can choke before this.